activerecord/test/cases/associations/eager_load_nested_include_test.rb

```diff
@@ -99,6 +102,25 @@ def test_include_query
       end
     end
   end
+
+  def test_lazy_preload_query
+    res = ShapeExpression.all.merge!(lazy_preload: [ :shape, { paint: :non_poly } ])
```

Comment:
This test covers only singular association. I think we need one for collection associations too, like: `Squre.all.lazy_preload(shape_expressions: { paint: :non_poly })`